src/main/java/seedu/address/logic/Logic.java

```diff
@@ -23,6 +24,7 @@ public interface Logic {
      */
     CommandResult execute(String commandText) throws CommandException, ParseException;
 
+    String commandType(String commandText) throws ParseException;
```

Comment:
Perhaps you could rename this method to getCommandType, a verb, I feel like this would make the function of the method more apparent.